File: frontend/src/microservices/content/objects/ReviewListPlannerResponse.ts
// ReviewListPlannerResponse：content 域评论列表返回对象。

import type { ReviewPlannerResponse } from './ReviewPlannerResponse'
export type ReviewListPlannerResponse = {
  reviews: ReviewPlannerResponse[]
}
export const reviewListPlannerResponseFromJson = (json: string): ReviewListPlannerResponse =>
  JSON.parse(json) as ReviewListPlannerResponse
export const reviewListPlannerResponseToJson = (value: ReviewListPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/ReviewListPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ReviewListPlannerResponse(reviews: List[ReviewPlannerResponse])
object ReviewListPlannerResponse:
  given sourceEncoder: Encoder[ReviewListPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ReviewListPlannerResponse] = deriveDecoder
